src/core/components/floodsub.js

```diff
@@ -0,0 +1,94 @@
+'use strict'
+
+const FloodSub = require('libp2p-floodsub')
+const promisify = require('promisify-es6')
+const Readable = require('stream').Readable
+
+const OFFLINE_ERROR = require('../utils').OFFLINE_ERROR
+const FSUB_ERROR = new Error(`FloodSub is not started.`)
+
+module.exports = function floodsub (self) {
+  return {
+    start: promisify((callback) => {
+      if (!self.isOnline()) {
+        throw OFFLINE_ERROR
+      }
+
+      self._floodsub = new FloodSub(self._libp2pNode)
+      return callback(null, self._floodsub)
```

Comment:
I believe it doesn't have to return the floodsub instance, as it gets available in the self object